src/TLuaInterpreterMudletObjects.cpp

```diff
@@ -2555,3 +2556,114 @@ int TLuaInterpreter::tempTrigger(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getProfiles
+int TLuaInterpreter::getProfiles(lua_State* L)
+{
+    auto& hostManager = mudlet::self()->getHostManager();
+    const QStringList profiles = QDir(mudlet::getMudletPath(mudlet::profilesPath))
+                                   .entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    lua_newtable(L);
+
+    for (const auto& profile : profiles) {
+        lua_pushstring(L, profile.toUtf8().constData());
+        lua_newtable(L);
+
+        QString url = mudlet::self()->readProfileData(profile, qsl("url"));
+        QString port = mudlet::self()->readProfileData(profile, qsl("port"));
+
+        // if url/port haven't been written to disk yet (which is what happens
+        // when a default profile is opened for the first time), fetch this data from game details
+        if (url.isEmpty()) {
+            auto it = TGameDetails::findGame(profile);
+            if (it != TGameDetails::scmDefaultGames.end()) {
+                url = (*it).hostUrl;
+            }
+        }
+        if (port.isEmpty()) {
+            auto it = TGameDetails::findGame(profile);
+            if (it != TGameDetails::scmDefaultGames.end()) {
+                port = QString::number((*it).port);
+            }
+        }
+
+        if (!url.isEmpty()) {
+            lua_pushstring(L, "host");
+            lua_pushstring(L, url.toUtf8().constData());
+            lua_settable(L, -3);
+        }
+
+        if (!port.isEmpty()) {
+            lua_pushstring(L, "port");
+            lua_pushstring(L, port.toUtf8().constData());
+            lua_settable(L, -3);
+        }
+
+        auto host = hostManager.getHost(profile);
+        const bool loaded = host != nullptr;
```

Comment:
Fair enough the second operator is a not-equal but two `=` signs on the same line with apparently equal visual presence I find a bit harder to parse - either put brackets around the comparison or perhaps this is clearer:
```suggestion
        const auto loaded = static_cast<bool>(host);
```